test/blob_test.js

```diff
@@ -332,36 +367,743 @@ suite.only("Azure Blob", function() {
+            assert(result.content === blockContent1.toString());
+
+            // commit another block
+            return blob.putBlock(containerName, blockBlobName, {
+              blockId: blockId2
+            }, blockContent2);
+          })
+          // verify that the block list contains an uncommitted block and a committed block
+            .then(function () {
+              return blob.getBlockList(containerName, blockBlobName, {
+                blockListType: 'all'
+              });
+            })
+            .then(function(result) {
+              assert(result.uncommittedBlocks.length === 1);
+              assert(result.uncommittedBlocks[0].blockId === blockId2);
+              assert(result.committedBlocks.length === 1);
+
+              return blob.putBlockList(containerName, blockBlobName, {
+                committedBlockIds: [result.committedBlocks[0].blockId],
+                uncommittedBlockIds: [result.uncommittedBlocks[0].blockId]
+              });
+            })
+            // verify that the blob is updated
+            .then(function (result) {
+              return blob.getBlob(containerName, blockBlobName);
+            })
+            .then(function (result) {
+              assert(result.content === (blockContent1.toString() + blockContent2));
+            })
+        });
+    });
+
+    test('create append blob', function() {
+      var options = {
+        blobType: 'AppendBlob'
+      };
+      return blob.putBlob(containerName, appendBlobName, options);
+    });
+
+    test('append block blob', function () {
+      var blobName = 'AppendBlob';
+      var content = Buffer.from("Lorem Ipsum is simply dummy text of the printing and typesetting industry. " +
+        "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a" +
+        " galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also" +
+        " the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the " +
+        "release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.");
+
+      return blob.appendBlock(containerName, appendBlobName, {}, content).then(function (result) {
+        assert(result.committedBlockCount === '1');
+      });
+    });
+
+    test('delete blob', function () {
+      return blob.deleteBlob(containerName, blockBlobName);
+    });
+
+    test('delete blob with if-match conditional header', function(){
+      var name = tempBlockBlobNamePrefix + '_if_match_conditional_header';
+      return blob.putBlob(containerName, name, {blobType: 'BlockBlob'}, 'Hello world')
+        .then(function (result) {
+          assert(result.eTag);
+
+          return blob.deleteBlob(containerName, name, {
```

Comment:
All the blobs are created in a container which has a name that starts with a prefix and, at the end, all the containers with that prefix are deleted, including all the created blobs. The conditional headers can be specified at the request level, but the information of ifMatch, ifModifiedSince, etc. is not stored in the properties of a blob (like the permissions are stored in the container properties). This means that, at cleanup time, the blobs will be deleted without any problems.